modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataListViewResourceImpl.java

```diff
@@ -319,8 +322,20 @@ private long _getClassNameId() {
 		return _portal.getClassNameId(DEDataListView.class);
 	}
 
-	private long _getDDMStructureId(DEDataListView deDataListView) {
-		return deDataListView.getDdmStructureId();
+	private DDMStructure _getDDMStructure(DEDataListView deDataListView) {
```

Comment:
@MrJohn1911 this is not ok. In the original logic the ddmStructureId was passed into the DataDefinitionModelResourcePermission.check() which internally did a DDMStructureLocalService.getDDMStructure() call. On non-existing id, it throws PortalException. But after your change it passes down a null to the new check method which internal just use the reference directly which will lead to a NPE.